pom.xml

```diff
@@ -81,23 +81,81 @@
     </snapshotRepository>
   </distributionManagement>
 
+
+  <properties>
+
+    <!-- Maven general -->
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+    <!-- maven-compiler-plugin -->
+    <maven.compiler.source>1.6</maven.compiler.source>
+    <maven.compiler.target>1.6</maven.compiler.target>
```

Comment:
The original motivation for 1.6 was the aim to get the java code hosted here to upstream, which was java 1.6 at that time. This already succeeded for license checker code, but is still to do for the checkstyle code. However, from the communication I had with checkstyle project lead, the chances seem to be low that they accept it  as they are focused on Java. maven-xml-plugin would perhaps be a better place for the code. Anyway, I agree that the compiler level should be revisited.